### PR TITLE
Explicitly validate the len of params to ensure we don't get index ou…

### PR DESCRIPTION
**Pull request by AndrewRPorter**
> Please make sure to open your PR against the `develop` branch.

v4 commits do not exist in `develop` branch, so PR is created against `main`.

## Pull Request Changes

- Explicitly validate the len of params to ensure we don't get index out of bounds errors. This fixes code scanning alerts like [CodeQL](https://codeql.github.com/).

## Closes

https://github.com/petergtz/pegomock/issues/125

### CODE REVIEW

**Review by petergtz — COMMENTED** [review]
I think this looks good. Can you change the PR from draft to "proper" PR? I hope this will trigger the CI.

**Comment by petergtz** [comment]
Ah, never mind my last comment, I was also able to remove the draft status.

**Comment by petergtz** [comment]
Okay, tests show this doesn't work for variadic arguments. I recommend to stick with changing from comparison `> 0` to `!= 0`, assuming it will satisfy CodeQL. This will not change the logic and therefore not break the tests.

**Comment by AndrewRPorter** [comment]
Sadly it doesn't look like changing the check to `!= 0` fixes the CodeQL warnings.

Instead, I just added a check for non variadic arguments.

This change generates verification methods like:

```go
func (c *MockDisplay_ChanParams_OngoingVerification) GetAllCapturedArguments() (_param0 []<-chan string, _param1 []chan<- error) {
	_params := pegomock.GetGenericMockFrom(c.mock).GetInvocationParams(c.methodInvocations)
	if len(_params) > 0 {
		if 0 < len(_params) {
			_param0 = make([]<-chan string, len(c.methodInvocations))
			for u, param := range _params[0] {
				_param0[u] = param.(<-chan string)
			}
		}
		if 1 < len(_params) {
			_param1 = make([]chan<- error, len(c.methodInvocations))
			for u, param := range _params[1] {
				_param1[u] = param.(chan<- error)
			}
		}
	}
	return
}
```

File: mockgen/mockgen.go [hunk 1]
```diff
@@ -365,10 +366,13 @@ func (g *generator) generateOngoingVerificationGetAllCapturedArguments(ongoingVe
 					p("}")
 				break
 			} else {
+				// explicitly validate the length of the params slice to avoid out of bounds code smells
+				g.p("if %v < len(_params) {", i)
```
Comment (petergtz) [inline mockgen/mockgen.go]: Could you please change this to:
```
g.p("if len(_params) > %v {", i)
```
Logically, it's the same, but it reads more naturally.

**Review by petergtz — COMMENTED** [review]
Okay, looks good and tests are green. Just one minor nit :-), then it's ready for merge.

File: mockgen/mockgen.go (hunk 1 above)
Comment (AndrewRPorter) [reply mockgen/mockgen.go]: Agreed, updated in https://github.com/petergtz/pegomock/pull/126/commits/54c3d6ebe5c03819d28b61444c6f80e9a6c1a061

**Comment by petergtz** [comment]
Merged and cut a new release 4.1.0 with your changes. Thank you very much for your contribution @AndrewRPorter.

**Pull request merged**